### PR TITLE
i.fft: added test file for i.fft module

### PR DESCRIPTION
**Pull request by jayneel-shah18**
This PR introduces a comprehensive suite of tests for the `i.fft` GRASS GIS module. These tests cover a wide range of scenarios, including basic functionality, advanced properties like energy conservation and linearity, and edge cases such as all-zero and all-one rasters. Performance for larger datasets is also assessed.  

Key updates in this PR include:  
- Validation of FFT output existence and dimensions.  
- Tests for linearity, energy conservation, and reconstruction via `i.ifft`.  
- Coverage of edge cases for specific raster patterns.  
- Performance assessment with large rasters.  

### Test Case Additions:  
1. **Basic Functionality**:  
   - Verify that `i.fft` produces real and imaginary output rasters.  
   - Ensure output dimensions match the input raster.  

2. **Advanced Properties**:  
   - **Linearity**: Combine two rasters and compare FFT results to individual components.  
   - **Energy Conservation**: Validate the energy conservation theorem.  
   - **Reconstruction**: Test `i.ifft` for accurate reconstruction of the original raster.  

3. **Edge Cases**:  
   - Behavior with an all-zero raster.  
   - Behavior with an all-one raster.  

4. **Performance**:  
   - Test FFT performance on a large 1000x1000 raster.

### CODE REVIEW

**Review by echoix — DISMISSED** [review]
Does someone else have something to add?

**Comment by wenzeslaus** [comment]
I think @petrasovaa had some comments, but I don't know if she can get to it.

File: imagery/i.fft/testsuite/test_i_fft.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+
+class TestIFFT(TestCase):
+    """Regression tests for the i.fft GRASS GIS module."""
+
+    input_raster = "test_input"
+    real_output = "fft_real"
+    imag_output = "fft_imag"
+
+    def setUp(self):
```
Comment (petrasovaa) [inline imagery/i.fft/testsuite/test_i_fft.py:14]: This should be probably run once, not for each test, see how other tests do it:
```
    @classmethod
    def setUpClass(cls):
...
    @classmethod
    def tearDownClass(cls):
```
https://stackoverflow.com/questions/23667610/what-is-the-difference-between-setup-and-setupclass-in-python-unittest

File: imagery/i.fft/testsuite/test_i_fft.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            imag_info["rows"],
+            "Mismatch in row count for imaginary raster",
+        )
+        self.assertEqual(
+            input_info["cols"],
+            imag_info["cols"],
+            "Mismatch in column count for imaginary raster",
+        )
+
+    def _parse_raster_values(self, raster):
```
Comment (petrasovaa) [inline imagery/i.fft/testsuite/test_i_fft.py:93]: Instead you can use gs.array API to get numpy array:
https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#module-script.array

File: imagery/i.fft/testsuite/test_i_fft.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        """Check that i.fft produces real and imaginary output rasters."""
+        self.assertModule(
+            "i.fft",
+            input=self.input_raster,
+            real=self.real_output,
+            imaginary=self.imag_output,
+            overwrite=True,
+        )
+        self.assertRasterExists(self.real_output)
+        self.assertRasterExists(self.imag_output)
```
Comment (petrasovaa) [inline imagery/i.fft/testsuite/test_i_fft.py:57]: You can test the output existence in the following tests, not needed to have a separate test for this, saves little bit of time.

File: imagery/i.fft/testsuite/test_i_fft.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+    def test_fft_output_dimensions(self):
```
Comment (petrasovaa) [inline imagery/i.fft/testsuite/test_i_fft.py:59]: This test is generally not needed, the dimensions are defined by the region.

File: imagery/i.fft/testsuite/test_i_fft.py (hunk 3 above)
Comment (echoix) [reply imagery/i.fft/testsuite/test_i_fft.py:57]: Keep in mind that currently, all the tests in this file take about 3 seconds on ci. It's nowhere near the minutes that the temporal tests take.

File: imagery/i.fft/testsuite/test_i_fft.py (hunk 1 above)
Comment (jayneel-shah18) [reply imagery/i.fft/testsuite/test_i_fft.py]: Thank you for the comment! I have made the changes and refactored the code to use `@classmethod` for `setUpClass` and `tearDownClass`, so the setup and teardown will now run once for the entire test class.

File: imagery/i.fft/testsuite/test_i_fft.py (hunk 2 above)
Comment (jayneel-shah18) [reply imagery/i.fft/testsuite/test_i_fft.py]: I have updated the code to use the `gs.array` API to get the numpy array and removed the helper function altogether.

File: imagery/i.fft/testsuite/test_i_fft.py (hunk 3 above)
Comment (jayneel-shah18) [reply imagery/i.fft/testsuite/test_i_fft.py]: I have removed the separate test for output existence as per the suggestion and now check for the existence of the output rasters in the subsequent tests themselves. This should help streamline the test suite.

File: imagery/i.fft/testsuite/test_i_fft.py (hunk 4 above)
Comment (jayneel-shah18) [reply imagery/i.fft/testsuite/test_i_fft.py]: I have removed this test as the dimensions are indeed defined by the region. I appreciate your input. Let me know if there is anything else you would like me to address.

**Review by petrasovaa — APPROVED** [review]
